### DISCUSSION TITLE
Dynamic field placement with text tags & an API upload question

### DISCUSSION THREAD

**Discussion by jspn-creative** [documenso/documenso#1328] [General]
Hello! I just came across this project and love everything about it, including the design focus, roadmap, transparency, etc.—Thank you for sharing it with the world!

There's a feature that I’m using with DocuSeal that my use case relies on, and I’m curious if it has been considered yet: text tags. 

Essentially, when uploading a PDF via api, I would like to include text tags in the pdf which would be parsed and replaced with document fields. I imagine it’s a feature that would eventually be implemented on some level in order to ship virtual documents—e.g. allowing me to dynamically specify where the receiver needs to sign and date and skipping the part where someone has to manually place those fields.

Examples of other implementations:
[DocuSeal](https://www.docuseal.co/guides/use-embedded-text-field-tags-in-the-pdf-to-create-a-fillable-form)
[Adobe Acrobat Sign](https://helpx.adobe.com/sign/using/text-tag.html)

Is there any chance that a feature like this is planned?

As an aside, I found the API documentation for uploading a document to be confusing—I don’t have any experience with pre-signed urls and actually conflated them with your direct link feature, thinking that a success response would contain a signing link for some reason. Do I understand correctly that after receiving a response with a pre-signed url, I’ll need to make a PUT request with the PDF in binary as the body—AKA as-is, and with a `Content-Type` of `application/pdf`? Would it be worth opening an issue to consider adding a small addendum to the developer docs to introduce the concept of pre-signed urls for file uploads?

Thank you again!

**Comment by kidhaxx** [comment]
Hi! 

I have the same need. I create quotes dynamically. The signature position changes with each quote (depending on the number of service lines, discounts, etc.).

Is it possible to position a field using a text anchor like this:
{{Name;type=text;role=SIGNER}}

**Comment by toddmzk** [comment]
depsarately need this feature too. It's available on Signnow which I've used in the past. Contract lengths depend on what text/content is inside them, so the positioning of signatures and fields is dependant on that also. We must have a way to dynamically set where fields go, when creating via API

**Comment by raldred** [comment]
We currently use Hellosign which has support for this. https://sfdc-docs.hellosign.com/docs/text-tags-syntax

Would be great to see sorted in documenso
